src/s3proxy/handlers/external.py

```diff
@@ -50,3 +53,33 @@ async def get_index(
         application_name=config.name,
     )
     return Index(metadata=metadata)
+
+
+@external_router.get(
+    "/s3/{bucket}/{key:path}",
+    description=(
+        "Return an S3 object's contents.  ``bucket`` can contain ``profile@``."
+    ),
+    summary="Object contents",
+)
+async def get_s3(
+    bucket: str,
+    key: str,
+    logger: Annotated[BoundLogger, Depends(auth_logger_dependency)],
+) -> Response:
+    """GET ``/s3proxy/s3/{bucket}/{key:path}``.
+
+    This returns the contents of an s3 object
+    """
+    path = f"s3://{bucket}/{key}"
+    logger.info("s3 request")
+    rp = ResourcePath(path)
+    mimetype, encoding = mimetypes.guess_type(path)
+    if mimetype is None:
+        mimetype = "application/octet-stream"
+    try:
+        return Response(content=rp.read(), media_type=mimetype)
```

Comment:
Are the objects always going to be small enough that we don't need to care about streaming them?

Reply:
DM-47720 filed for this.  For now, everything should fit just fine.